addons/website_sale/controllers/main.py

```diff
@@ -2125,7 +2125,11 @@ def _get_payment_values(self, order_sudo, website_id=None, **kwargs):
         :return: The payment-specific values.
         :rtype: dict
         """
-        website_id = website_id or order_sudo.website_id.id
+        if order_sudo.website_id:
+            website_id = order_sudo.website_id.id
+        elif request.website:
+            website_id = request.website.id
+
```

Comment:
Please keep the provided `website_id` if set, it should only be replaced by the order/request website if not set.